File: CTask/src/org/ShapeSHifter420/Fabric/LocationPushFactory.java
package org.ShapeSHifter420.Fabric;

import org.ShapeSHifter420.Pushes.IPush;
import org.ShapeSHifter420.Pushes.LocationPush;

import java.util.Map;

public class LocationPushFactory implements IPushFactory{
    @Override
    public IPush create(Map<String, String> conf) {
        String text = conf.getOrDefault("text", "");
        long exp = conf.containsKey("expiry_date")? Long.parseLong(conf.get("expiry_date")): -1;
        float xcoord = conf.containsKey("x_coord")? Float.parseFloat(conf.get("x_coord")) : 0;
        float ycoord = conf.containsKey("y_coord")? Float.parseFloat(conf.get("y_coord")) : 0;
        float radius = conf.containsKey("radius")? Float.parseFloat(conf.get("radius")) : -1;
        return new LocationPush(text,exp,xcoord,ycoord,radius);
    }
}

File: CKotlin/Factory/LocationPushFactory.kt
package Factory

import Pushes.IPush
import Pushes.LocationPush

object LocationPushFactory : IPushFactory {
    override fun made(param: Map<String, String>): IPush = LocationPush(
        xCoord = param["x_coord"]?.toDouble() ?: 0.0,
        yCoord = param["y_coord"]?.toDouble() ?: 0.0,
        radius = param["radius"]?.toDouble() ?: 0.0,
        text = param["text"] ?: "none",
        expiry = param["expiry_date"]?.toLong() ?: 0,
    )
}